kernel/kernel-api/src/test/scala/io/delta/kernel/internal/SnapshotManagerSuite.scala

```diff
@@ -775,31 +776,43 @@ class SnapshotManagerSuite extends AnyFunSuite with MockFileSystemClientUtils {
 }
 
 trait SidecarIteratorProvider extends VectorTestUtils {
-  def singletonSidecarIterator(sidecars: Seq[FileStatus])
-      : CloseableIterator[ColumnarBatch] = Utils.singletonCloseableIterator(
-    new ColumnarBatch {
-      override def getSchema: StructType = SidecarFile.READ_SCHEMA
-
-      override def getColumnVector(ordinal: Int): ColumnVector = {
-        ordinal match {
-          case 0 => stringVector(sidecars.map(_.getPath)) // path
-          case 1 => longVector(sidecars.map(_.getSize): _*) // size
-          case 2 =>
-            longVector(sidecars.map(_.getModificationTime): _*); // modification time
-        }
-      }
 
-      override def getSize: Int = sidecars.length
-    })
+  // Shared helper to construct the batch
+  private def buildSidecarBatch(sidecars: Seq[FileStatus]): ColumnarBatch = new ColumnarBatch {
+    override def getSchema: StructType = SidecarFile.READ_SCHEMA
+
+    override def getColumnVector(ordinal: Int): ColumnVector = ordinal match {
+      case 0 => stringVector(sidecars.map(_.getPath)) // path
+      case 1 => longVector(sidecars.map(_.getSize): _*) // size
+      case 2 => longVector(sidecars.map(_.getModificationTime): _*) // modification time
+    }
+
+    override def getSize: Int = sidecars.length
+  }
+
+  // For Parquet (ColumnarBatch + filePath)
+  def singletonSidecarParquetIterator(sidecars: Seq[FileStatus])
+      : CloseableIterator[FileReadResult] = {
+    val batch = buildSidecarBatch(sidecars)
+    val filePath = if (sidecars.nonEmpty) sidecars.head.getPath else "<no_file>"
```

Comment:
so we are mocking reading these sidecar files from a v2 parquet checkpoint manifest, right?

`10.uuid.checkpoint.parquet` ?

so shouldn't the "file path" be the checkpoint file path, and not the file path of the inner contents of that parquet file? i.e. i don't think it should be the first sidecar path? for example, why not the 2nd or 3rd sidecar path?